packages/jest-snapshot/src/index.js

```diff
@@ -61,12 +66,43 @@ const toMatchSnapshot = function(received: any, testName?: string) {
     throw new Error('Jest: snapshot state must be initialized.');
   }
 
-  const result = snapshotState.match(
+  const fullTestName =
     testName && currentTestName
       ? `${currentTestName}: ${testName}`
-      : currentTestName || '',
-    received,
-  );
+      : currentTestName || '';
+
+  if (typeof propertyMatchers === 'object') {
+    const propertyPass = this.equals(received, propertyMatchers, [
+      this.iterableEquality,
+      this.subsetEquality,
+    ]);
+
+    if (!propertyPass) {
+      const key = snapshotState.fail(fullTestName, received);
+
+      const report = () =>
+        `${RECEIVED_COLOR('Received value')} does not match ` +
+        `${EXPECTED_COLOR(`snapshot properties for "${key}"`)}.\n\n` +
+        `Expected snapshot to match properties:\n` +
+        `  ${this.utils.printExpected(propertyMatchers)}` +
+        `\nReceived:\n` +
+        `  ${this.utils.printReceived(received)}`;
+
+      return {
+        message: () =>
+          matcherHint('.toMatchSnapshot', 'value', 'properties') +
+          '\n\n' +
+          report(),
+        name: 'toMatchSnapshot',
+        pass: false,
+        report,
+      };
+    } else {
+      Object.assign(received, propertyMatchers);
```

Comment:
This won't handle deep objects - should it be a deep merge instead?

Reply:
Great catch, i'll add an integration test for this as well

Reply:
Any follow up for this?

Reply:
#6528